### ISSUE TITLE
sink._addParentTeardownLogic is not a function rxjs node 6.3.2 and 6.2.2

### ISSUE THREAD

**Issue by alan-agius4** [ReactiveX/rxjs#4135]
## Bug Report

**Current Behavior**
```
sink._addParentTeardownLogic is not a function
```
When using rxjs 6.2.2 and 6.3.2.

npm list rxjs  
```
rxjs-tear-down-logic@1.0.0 
+-- fakes-dependency@1.0.0
| `-- rxjs@6.3.2
`-- rxjs@6.2.2
```

**Reproduction**
- https://github.com/alan-agius4/rxjs-teardown-logic-issue
- npm i 
- npm start

rxjs 6.3.2
```js
const rxjs = require('rxjs');

const result = new rxjs.Observable(obs => {
	return rxjs.of(undefined).subscribe(obs)
})

module.exports = { result: result }
```

rxjs 6.2.2
```js
const rxjs = require('rxjs');
const dependency = require('fakes-dependency');
const rxjs_operators = require('rxjs/operators');

rxjs.of(undefined).pipe(
	rxjs_operators.concatMap(() => dependency.result),
).toPromise()

```

**Expected behavior**
No Error

**Environment**
- Runtime: Node v 8.9
- RxJS version: 6.2.2 and 6.3.2

**Comment by benlesh** [comment]
cc @cartant ... This is an issue with identifying a "trusted subscriber" between two different versions of RxJS. Basically one RxJS's Subscriber gets a Subscriber from the other RxJS (which doesn't have `_addParentTeardownLogic`)

This is directly related to our discussion yesterday that `_addParentTeardownLogic` is probably the wrong solution and needs to be removed/replaced.

**Comment by benlesh** [comment]
#4106 is related to this. It should help resolve the issue.

**Comment by alan-agius4** [comment]
Confirmed that it has been resolved in 6.3.3
